File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: list[int], x: int) -> int:
        total = sum(nums)
        target = total - x

        if target == 0:
            return len(nums)

        if target < 0:
            return -1

        left = 0
        window_sum = 0
        window_size = -1

        for right in range(len(nums)):
            window_sum += nums[right]

            while window_sum > target:
                window_sum -= nums[left]
                left += 1

            if window_sum == target:
                window_size = max(window_size, right - left + 1)

        return -1 if window_size == -1 else len(nums) - window_size

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.js
/**
 * @param {number[]} nums
 * @param {number} x
 * @return {number}
 */
var minOperations = function (nums, x) {
    const total = nums.reduce((sum, curr) => sum + curr, 0);
    const target = total - x;

    if (target === 0) {
        return nums.length;
    }

    if (target < 0) {
        return -1;
    }


    let left = 0;
    let sum = 0;
    let windowSize = -1;

    for (let right = 0; right < nums.length; right++) {
        sum += nums[right];

        while (sum > target) {
            sum -= nums[left];
            left++;
        }

        if (sum === target) {
            windowSize = Math.max(windowSize, right - left + 1);
        }
    }

    return windowSize === -1 ? -1 : nums.length - windowSize;
};